abi/encode.go

```diff
@@ -193,6 +201,13 @@ func encodeBytes(v reflect.Value) ([]byte, error) {
 	if v.Kind() == reflect.Array {
 		v = convertArrayToBytes(v)
 	}
+	if v.Kind() == reflect.String {
+		value, err := decodeHex(v.String())
+		if err != nil {
+			return nil, err
+		}
+		return packBytesSlice(value, len(value))
```

Comment:
Could you do this instead, so we can reuse the `packBytesSlice` call?
```suggestion
		v = value
```

Reply:
It cannot since `v` is expected as `reflect.Value` and `value` is a `[]byte`. I would do the mapping but IMO that would be an unnecessary overhead in the code. WDYT?

Reply:
You are right on that, I like your approach better.

Reply:
Hi @ferranbt if you refresh you should already see that change applied in my PR. 

PD: We are waiting for them to be merged in master to fix an issue on our software affecting ours users.

Reply:
Ok, as soon as CI is done I merge it